### ISSUE TITLE
Add reverse proxy for http3

### ISSUE THREAD

**Issue by yuwenlong** [nginx/nginx#376]
### Describe the feature you'd like to add to nginx

Add reverse proxy for http3

### Describe the problem this feature solves

proxy_pass does not support http3, hope it does! thank you

**Comment by HanadaLee** [comment]
refer to https://mailman.nginx.org/pipermail/nginx-devel/2023-December/THMZAQ36SN5BICJSCLX6FLEUI45FHR4H.html

**Comment by yuwenlong** [comment]
@HanadaLee 
proxy_http_version 3;
Error: nginx: [emerg] invalid value "3"

nginx version: nginx/1.25.5
built by gcc 13.2.0 (Ubuntu 13.2.0-4ubuntu3)
built with OpenSSL 3.3.0 9 Apr 2024
TLS SNI support enabled
configure arguments: --with-http_ssl_module --with-http_v2_module --with-http_v3_module --with-stream --with-stream_ssl_module --with-stream_ssl_preread_module --with-openssl=/opt/openssl --add-module=/opt/nginx_module/tengine/modules/ngx_http_concat_module --add-module=/opt/nginx_module/ngx_devel_kit --add-module=/opt/nginx_module/lua-nginx-module --add-module=/opt/nginx_module/echo-nginx-module --add-module=/opt/nginx_module/ngx_http_proxy_connect_module --add-module=/opt/nginx_module/stream-lua-nginx-module --with-ld-opt=-Wl,-rpath,/usr/local/luajit/lib

**Comment by HanadaLee** [comment]
> [@HanadaLee](https://github.com/HanadaLee) proxy_http_version 3; Error: nginx: [emerg] invalid value "3"
> 
> nginx version: nginx/1.25.5 built by gcc 13.2.0 (Ubuntu 13.2.0-4ubuntu3) built with OpenSSL 3.3.0 9 Apr 2024 TLS SNI support enabled configure arguments: --with-http_ssl_module --with-http_v2_module --with-http_v3_module --with-stream --with-stream_ssl_module --with-stream_ssl_preread_module --with-openssl=/opt/openssl --add-module=/opt/nginx_module/tengine/modules/ngx_http_concat_module --add-module=/opt/nginx_module/ngx_devel_kit --add-module=/opt/nginx_module/lua-nginx-module --add-module=/opt/nginx_module/echo-nginx-module --add-module=/opt/nginx_module/ngx_http_proxy_connect_module --add-module=/opt/nginx_module/stream-lua-nginx-module --with-ld-opt=-Wl,-rpath,/usr/local/luajit/lib

It has not been merged into mainline nginx yet, it is just a concept patch. Maybe you can try it first with Angie (a fork of nginx), which already supports

**Comment by yuwenlong** [comment]
@HanadaLee Are you an official? When will the official support come?

**Comment by HanadaLee** [comment]
> [@HanadaLee](https://github.com/HanadaLee) Are you an official? When will the official support come?

No, I am not an official. If you need to know the plan, you can wait for the official to reply to you.

**Comment by colben** [comment]
I've been looking forward this feature(proxy_pass quic) for several years!
Yestarday i tested this expermental feature in caddy, it worked!
Here is my "Caddyfile":
```
http://:8888 {
    reverse_proxy https://my-nginx-quic-server:7443 {
        transport http {
            versions 3
        }
    }
}
```
But, there is an unexpected error: the first request after idle 60s always gets a respond "502 Bad Gatewat", and it will work again when i send another request in a few seconds.
